### ISSUE TITLE
IEnumerator.ToUniTask() doesn't behave like how Unity StartCoroutine behaves

### ISSUE THREAD

**Issue by Lf3T-Hn4D** [Cysharp/UniTask#120]
When calling StartCoroutine(), Unity will execute the coroutine until the first yield instruction. ToUniTask() currently does not behave the same way. It fully relies on the player loop helper to execute the coroutine. This causes issue with coroutine codes that relies on coroutines to be executed immediately until first yield instruction.

I was able to change the behaviour by adding call to MoveNext() in EnumeratorPromise.Create() like so:
```csharp
            public static IUniTaskSource Create(IEnumerator innerEnumerator, PlayerLoopTiming timing, CancellationToken cancellationToken, out short token)
            {
                if (cancellationToken.IsCancellationRequested)
                {
                    return AutoResetUniTaskCompletionSource.CreateFromCanceled(cancellationToken, out token);
                }

                if (!pool.TryPop(out var result))
                {
                    result = new EnumeratorPromise();
                }
                TaskTracker.TrackActiveTask(result, 3);

                result.innerEnumerator = ConsumeEnumerator(innerEnumerator);
                result.cancellationToken = cancellationToken;

                PlayerLoopHelper.AddAction(timing, result);

                token = result.core.Version;

                // Make coroutine integration behave like Unity's where first section of the coroutine is executed immediately.
                result.MoveNext();
                return result;
            }
```

Is this an overlooked feature or is there some known problem that such behaviour was omitted? If there's no side effect to doing this, but isn't a desired default behaviour, I'd like to have an option to have a choice of executing the coroutine immediately.

**Comment by neuecc** [comment]
Thanks, 
I would like to fix it so that it behaves the same as StartCoroutine, as you say.

It's important to note that StartCoroutine(yield return null) does not fire twice on the same frame.
Therefore, I need to adjust IPlayerLoopItem.MoveNext by getting Time.frameCount.

**Comment by Lf3T-Hn4D** [comment]
Great! Thanks. And yes, I totally missed the case of the double execution in the same frame. Good catch 👍🏼

**Comment by neuecc** [comment]
I've released `2.0.28`, it includes this fix.
